backend/data_adapter.go

```diff
@@ -17,7 +19,26 @@ func newDataSDKAdapter(handler QueryDataHandler) *dataSDKAdapter {
 	}
 }
 
+func withHeaderMiddleware(ctx context.Context, headers map[string]string) context.Context {
+	if len(headers) > 0 {
+		ctx = httpclient.WithContextualMiddleware(ctx,
+			httpclient.MiddlewareFunc(func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
+				return httpclient.RoundTripperFunc(func(qreq *http.Request) (*http.Response, error) {
+					// Only set a header if it is not already set.
+					for k, v := range headers {
+						if qreq.Header.Get(k) == "" {
+							qreq.Header.Set(k, v)
+						}
+					}
+					return next.RoundTrip(qreq)
+				})
+			}))
+	}
+	return ctx
+}
+
 func (a *dataSDKAdapter) QueryData(ctx context.Context, req *pluginv2.QueryDataRequest) (*pluginv2.QueryDataResponse, error) {
+	ctx = withHeaderMiddleware(ctx, req.Headers)
```

Comment:
Please use `req.GetHTTPHeaders()` in adapters. You'll need to convert from protobuf before this method is available on the converted request, see https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go@v0.149.1/backend#QueryDataRequest.GetHTTPHeaders